File: shiftheroes_request.py
import requests
import os

from dotenv import load_dotenv

load_dotenv()

headers = {
    'Authorization': f'Bearer {os.getenv("API_KEY")}',
}

# Récupérer la liste des plannings
response = requests.get('https://shiftheroes.fr/api/v1/plannings', headers=headers)
# Affiche la réponse JSON convertie en objet Python
print(response.json())

# Récupérer la liste des crénaux d'un planning
response = requests.get('https://shiftheroes.fr/api/v1/plannings/LQfrZg/shifts', headers=headers)
# Affiche la réponse JSON convertie en objet Python
print(response.json())

# Récupérer une réservation sur un crénau
response = requests.post('https://shiftheroes.fr/api/v1/plannings/LQfrZg/shifts/OAF6DPq/reservations', headers=headers)
# Affiche la réponse JSON convertie en objet Python
print(response.json())

File: shiftheroes_request.swift
import Foundation

let url = URL(string: "https://shiftheroes.fr/api/v1/plannings/LQfrZg/shifts/OAF6DPq/reservations")!
let headers = [
    "Authorization": "Bearer 376f0a25129fda22f7fc44398757efc7"
]

var request = URLRequest(url: url)
request.httpMethod = "POST"
request.allHTTPHeaderFields = headers

let task = URLSession.shared.dataTask(with: request) { (data, response, error) in
    if let error = error {
        print(error)
    } else if let data = data {
        let str = String(data: data, encoding: .utf8)
        print(str ?? "")
    }
}

task.resume()